bin/build-webpack.sh

```diff
@@ -0,0 +1,138 @@
+# Functions
+# --------------------------------------
+
+#
+# Print script usage
+#
+usage() {
+  if [[ $1 -ne 0 ]]; then
+    exec 1>&2
+  fi
+
+  echo -e ""
+  echo -e "Usage:\t$SCRIPT_NAME [-h] [-s SHIP_NAME] [-u URL]"
+  echo -e ""
+  echo -e "Build the app frontend and the desk files required to install it in Grid"
+  echo -e ""
+  echo -e "Options:"
+  echo -e "  -h\tPrint script usage info"
+  echo -e "  -s\tSet ship name to use (default: $DEFAULT_SHIP)"
+  echo -e "  -u\tUse given URL to distribute glob over HTTP instead of over Ames"
+  echo -e ""
+  exit $1
+}
+
+#
+# Build docket file
+#
+docket() {
+  DOCKET_FILE=$DESK_DIR/desk.docket-0
+
+  echo ":~" > $DOCKET_FILE
+  echo "  title+'Chess'" >> $DOCKET_FILE
+  echo "  info+'Fully peer-to-peer chess over Urbit'" >> $DOCKET_FILE
+  echo "  color+0xff.ffff" >> $DOCKET_FILE
+  echo "  image+'https://peekabooo.icu/images/finmep-chess.svg'" >> $DOCKET_FILE
+  echo "  base+'chess'" >> $DOCKET_FILE
+  echo "  version+[$VERSION_MAJOR $VERSION_MINOR $VERSION_PATCH]" >> $DOCKET_FILE
+  echo "  license+'GPL3'" >> $DOCKET_FILE
+  echo "  website+'https://github.com/ashelkovnykov/urbit-chess'" >> $DOCKET_FILE
+  
+  if [[ -z $URL ]]; then
+    echo "  glob-ames+[~$SHIP 0v0]" >> $DOCKET_FILE
+  else
+    GLOB=$(echo ${URL} | grep -Eo '0v[0-9a-v]{1,5}(\.[0-9a-v]{5})+')
+    echo "  glob-http+['$URL' $GLOB]" >> $DOCKET_FILE
+  fi
+  
+  echo "==" >> $DOCKET_FILE
+}
+
+# --------------------------------------
+# Variables
+# --------------------------------------
+
+SCRIPT_NAME=$(basename $0 | cut -d '.' -f 1)
+SCRIPT_PATH=$(realpath $0)
+SCRIPT_DIR=$(dirname $SCRIPT_PATH)
+
+ROOT_DIR=$(dirname $SCRIPT_DIR)
+BUILD_DIR="$ROOT_DIR/build"
+DESK_DIR="$BUILD_DIR/desk"
+FRONTEND_DIR="$BUILD_DIR/frontend"
+
+DOCKER_IMAGE="urbit-chess"
+
+VERSION_MAJOR=0
+VERSION_MINOR=9
+VERSION_PATCH=1
+VERSION_FULL="$VERSION_MAJOR.$VERSION_MINOR.$VERSION_PATCH"
+
+KELVIN=418
+
+DEFAULT_SHIP="finmep-lanteb"
+SHIP=$DEFAULT_SHIP
+
+# --------------------------------------
+# MAIN
+# --------------------------------------
+
+# Parse arguments
+OPTS=":hs:u:"
+while getopts ${OPTS} opt; do
+  case ${opt} in
+    h)
+      usage 0
+      ;;
+    s)
+      SHIP=$OPTARG
+      ;;
+    u)
+      URL=$OPTARG
+      ;;
+    :)
+      echo "$SCRIPT_NAME: Missing argument for '-${OPTARG}'" >&2
+      usage 2
+      ;;
+    ?)
+      echo "$SCRIPT_NAME: Invalid option '-${OPTARG}'" >&2
+      usage 2
+      ;;
+  esac
+done
+
+# Clean up dirs before running
+rm -rf $BUILD_DIR
+mkdir -p $DESK_DIR
+
+# Build desk.bill
+echo ":~  %chess  ==" > $DESK_DIR/desk.bill
+
+# Build desk.docket-0
+docket
+
+# Build desk.ship
+echo "~$SHIP" > $DESK_DIR/desk.ship
+
+# Build sys.kelvin
+echo "[%zuse $KELVIN]" > $DESK_DIR/sys.kelvin
+
+# Build frontend
+# sudo docker build --tag ${DOCKER_IMAGE}:${VERSION_FULL} .
+# sudo docker run --rm -v ${FRONTEND_DIR}:/app/output/ ${DOCKER_IMAGE}:${VERSION_FULL}
+(cd "$ROOT_DIR/src/frontend"; npm run build-no-docker)
```

Comment:
Looks like this is the only real change. So that we don't have two near identical scripts, each of which must have a correctly set version, let's replace this script with a `--no-docker` option in the regular build script:
```
...

usage() {
  if [[ $1 -ne 0 ]]; then
    exec 1>&2
  fi

  echo -e ""
  echo -e "Usage:\t$SCRIPT_NAME [-h] [-n] [-s SHIP_NAME] [-u URL]"
  echo -e ""
  echo -e "Build the app frontend and the desk files required to install it in Grid"
  echo -e ""
  echo -e "Options:"
  echo -e "  -h\tPrint script usage info"
  echo -e "  -n\tUse npm natively instead of through Docker"
  echo -e "  -s\tSet ship name to use (default: $DEFAULT_SHIP)"
  echo -e "  -u\tUse given URL to distribute glob over HTTP instead of over Ames"
  echo -e ""
  exit $1
}

...

DOCKER=1
DOCKER_IMAGE="urbit-chess"

...


OPTS=":hs:u:"
while getopts ${OPTS} opt; do
  case ${opt} in
    h)
      usage 0
      ;;
    n)
      DOCKER=0
    s)
      SHIP=$OPTARG
      ;;
    u)
      URL=$OPTARG
      ;;
    :)
      echo "$SCRIPT_NAME: Missing argument for '-${OPTARG}'" >&2
      usage 2
      ;;
    ?)
      echo "$SCRIPT_NAME: Invalid option '-${OPTARG}'" >&2
      usage 2
      ;;
  esac
done

...

# Build frontend
if [ $DOCKER -eq 1 ]; then
  sudo docker build --tag ${DOCKER_IMAGE}:${VERSION_FULL} .
  sudo docker run --rm -v ${FRONTEND_DIR}:/app/output/ ${DOCKER_IMAGE}:${VERSION_FULL}

  # Copy additional src files for frontend
  sudo chown -R ${USER}:${USER} ${FRONTEND_DIR}
else
  (cd "$ROOT_DIR/src/frontend"; npm run build-no-docker)
fi
```